src/tensor_ops/tri.rs

```diff
@@ -0,0 +1,110 @@
+use crate::shapes::{Dtype, Shape};
+use crate::tensor::{HasErr, Tape, Tensor, TriangleTensor};
+
+use super::TryMul;
+
+pub fn lower_tri<S: Shape, E: Dtype, D: TriangleTensor<E>, T: Tape<E, D>>(
```

Comment:
Will need to put a docstring on all these functions/methods (lower_tri/upper_tri & the tensor methods of each). We can make this a separate documentation issue if you don't want to address in this PR